src/Symfony/Bundle/FrameworkBundle/Command/RouterDebugCommand.php

```diff
@@ -115,4 +121,35 @@ private function findRouteNameContaining(string $name, RouteCollection $routes):
 
         return $foundRoutesNames;
     }
+
+    private function filterRoutes(RouteCollection $routes, InputInterface $input)
+    {
+        if (!empty($methods = $input->getOption('method'))) {
+            $routes = $routes->filter(function (Route $route) use ($methods) {
+                return \array_intersect($methods, $route->getMethods());
+            });
+        }
+
+        if (null !== ($scheme = $input->getOption('scheme'))) {
+            if (!\in_array($scheme, array('http', 'https'))) {
+                throw new InvalidArgumentException(\sprintf('The "%s" scheme is not valid. Valid schemes are "http" and "https".', $scheme));
```

Comment:
Why is there a validation on the passed scheme and not on method ? AFAIK, you can define any method and scheme in a Route.